bool_test.go

```diff
@@ -0,0 +1,38 @@
+package option_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vmihailenco/msgpack/v5"
+
+	"github.com/tarantool/go-option"
+)
+
+func TestOptionalBool(t *testing.T) {
+	t.Parallel()
+
+	var buf bytes.Buffer
+
+	enc := msgpack.NewEncoder(&buf)
+	dec := msgpack.NewDecoder(&buf)
+
+	someBool := option.SomeBool(true)
+	assert.True(t, someBool.IsSome(), "Expected IsPresent() to return true")
+	require.NoError(t, someBool.EncodeMsgpack(enc))
+
+	var shouldBeSomeBool option.Bool
+	require.NoError(t, shouldBeSomeBool.DecodeMsgpack(dec))
+	assert.True(t, shouldBeSomeBool.IsSome(), "Expected IsPresent() to return true")
+	assert.Equal(t, someBool.Unwrap(), shouldBeSomeBool.Unwrap(), "Expected Value() to return the same value")
+
+	emptyBool := option.NoneBool()
+	assert.False(t, emptyBool.IsSome(), "Expected IsPresent() to return false")
+	require.NoError(t, emptyBool.EncodeMsgpack(enc))
+
+	var shouldBeEmptyBool option.Bool
+	require.NoError(t, shouldBeEmptyBool.DecodeMsgpack(dec))
+	assert.False(t, shouldBeEmptyBool.IsSome(), "Expected IsPresent() to return false")
+}
```

Comment:
Please, split the test into 3 or 4 tests by AAA-pattern:

https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

Here we have a one tests for multiple test cases. This is a bad example for future developers.

Reply:
It looks like we don't cover all methods in the tests. MustGet as example.